resources/testharness.js

```diff
@@ -1934,9 +2032,33 @@ policies and contribution forms [3].
     };
 
     Tests.prototype.timeout = function() {
+        var cleaningTest = null;
+
         if (this.status.status === null) {
-            this.status.status = this.status.TIMEOUT;
+            forEach(this.tests,
+                    function(test) {
+                        if (test.phase === test.phases.CLEANING) {
+                            cleaningTest = test;
+                        }
+
+                        test.phase = test.phases.COMPLETE;
+                    });
+
+            // Timeouts that occur while a sub-test is in the "cleanup" phase
+            // indicate that some global state was not properly reverted. This
+            // invalidates the overall test execution, so the timeout should be
+            // reported as an error and cancel the execution of any remaining
+            // sub-tests.
+            if (cleaningTest) {
+                this.status.status = this.status.ERROR;
+                this.status.message = "Cleanup function for test named \"" +
```

Comment:
The cleanup function itself doesn't have a timeout, so maybe "Timeout while running cleanup for test named ..."?

Reply:
Oooh, that is a really nuanced distinction. I like your text much better!